### PR TITLE
Feature/error message fetching#169

### LINKED ISSUE #169: (SP: 2 ) show error message to User if an error occurs while fetching categories in the SubjectsStep component
Description:

When fetching categories in the SubjectsStep and General step component, if an error occurs, the current implementation logs the error to the console. To enhance user experience, it would be beneficial to display an error message instead of logging to the console.

Tasks:
Display Error to User: When an error occurs during the fetching of categories, update the UI to show the error message.

Testing: Write tests to verify that the error handling works as expected.

### PR DESCRIPTION
**Pull request by marianamartsy**
close #169 

Show error message to User if an error occurs while fetching categories in the SubjectsStep component
When fetching categories in the SubjectsStep and name, country and city in General step component, if an error occurs, the current implementation logs the error to the console. 

tests coverage:
Zrzut ekranu 2023-12-13 o 23 10 45

- [x] Display Error to User: When an error occurs during the fetching of categories, update the UI to show the error message.

- [x] Write tests to verify that the error handling works as expected.
Zrzut ekranu 2023-12-13 o 20 03 48
Zrzut ekranu 2023-12-13 o 20 04 05

Zrzut ekranu 2023-12-13 o 19 20 04

### CODE REVIEW

**Review by LeonidShamarin — APPROVED** [review]

**Review by Mygal-Alex — APPROVED** [review]

**Review by ArtemHolikov — APPROVED** [review]
👍

**Review by AndriiNasadiuk — APPROVED** [review]

File: src/constants/translations/en/common.json [hunk 1]
```diff
@@ -55,7 +55,8 @@
     "shortText": "This text is too short",
     "longText": "This text is too long",
     "numbersOnly": "Please enter a valid number",
-    "positiveNumbersOnly": "Please enter a positive number"
+    "positiveNumbersOnly": "Please enter a positive number",
+    "fetchingData": "Failed to upload data. Please try again."
```
Comment (vlad-khrychov) [inline src/constants/translations/en/common.json:59]: Please create an additional task in backlog to translate error messages and other strings into Ukrainian.

File: tests/unit/containers/tutor-home-page/subject-step/SubjectStep.spec.jsx [hunk 2]
```diff
@@ -2,6 +2,16 @@ import { render, fireEvent, screen, waitFor } from '@testing-library/react'
...
+import { useSnackBarContext } from '~/context/snackbar-context'
+
+vi.mock('~/context/snackbar-context', () => ({
+  useSnackBarContext: vi.fn(() => ({
+    setAlert: vi.fn().mockReturnValue({
+      severity: 'error',
+      message: 'common.errorMessages.fetchingData'
+    })
+  }))
+}))
```
Comment (vlad-khrychov) [inline tests/unit/containers/tutor-home-page/subject-step/SubjectStep.spec.jsx:14]: @ArtemHolikov here you can find mock of object where one of properties is mocked function (`setAlert`)

File: tests/unit/containers/tutor-home-page/subject-step/SubjectStep.spec.jsx [hunk 3]
```diff
@@ -77,46 +95,162 @@ describe('SubjectsStep component test', () => {
...
+        message: 'common.errorMessages.fetchingData'
+      })
+    })
+  })
+  it('should use setAlert from useSnackBarContext', () => {
+    expect(useSnackBarContext).toHaveBeenCalled()
+
+    const { setAlert } = useSnackBarContext()
+
+    expect(typeof setAlert).toBe('function')
```
Comment (vlad-khrychov) [inline tests/unit/containers/tutor-home-page/subject-step/SubjectStep.spec.jsx]: We already have tests on calling `setAlert` so, i think we can delete it from here.

**Review by vlad-khrychov — APPROVED** [review]

File: tests/unit/containers/tutor-home-page/subject-step/SubjectStep.spec.jsx (hunk 3 above)
Comment (marianamartsy) [reply tests/unit/containers/tutor-home-page/subject-step/SubjectStep.spec.jsx]: 👍

File: src/constants/translations/en/common.json (hunk 1 above)
Comment (marianamartsy) [reply src/constants/translations/en/common.json:59]: 👍

File: tests/unit/components/user-steps-wraper/UserStepsWrapper.spec.jsx [hunk 4]
```diff
@@ -13,9 +13,45 @@ vi.mock('react-redux', async () => {
...
+  )
+}))
+
+vi.mock('~/redux/reducer', () => ({
+  markFirstLoginComplete: vi.fn()
+}))
+
+vi.mock('~/context/step-context', () => ({
+  StepProvider: ({ children }) => <div>{children}</div>
+}))
```
Comment (online-marathon) [inline tests/unit/components/user-steps-wraper/UserStepsWrapper.spec.jsx:29]: @Mygal-Alex here you can find mock of provider

**Review by online-marathon — APPROVED** [review]

**Pull request merged**